purchase_stock_return_request/models/stock_return_request.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class StockReturnRequest(models.Model):
+    _inherit = 'stock.return.request'
+
+    purchase_order_ids = fields.Many2many(
+        comodel_name='purchase.order',
+        string='Involved Purchases',
+        readonly=True,
+        copy=False,
+    )
+
+    def _prepare_move_default_values(self, line, qty, move):
+        """Extend this method to add values to return move"""
+        vals = super()._prepare_move_default_values(line, qty, move)
+        vals.update({
+            'purchase_line_id': move.purchase_line_id.id,
+            'to_refund': line.request_id.to_refund,
+        })
+        return vals
+
+    def _action_confirm(self):
+        res = super()._action_confirm()
+        if self.state == 'confirmed':
+            self.purchase_order_ids = self.returned_picking_ids.mapped(
+                'purchase_id')
+        return res
+
+    def action_view_purchases(self):
+        """Display returned purchases"""
+        action = action = self.env.ref('purchase.purchase_form_action')
```

Comment:
Why this chained assignment? Did you mean this?
```suggestion
        action = self.env.ref('purchase.purchase_form_action')
```